pkg/query/rules.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package query
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+)
+
+func NewRulesRetriever(rs storepb.RulesServer) *rulesRetriever {
+	return &rulesRetriever{
+		proxy: rs,
+	}
+}
+
+type rulesRetriever struct {
+	proxy storepb.RulesServer
+}
+
+func (rr *rulesRetriever) RuleGroups(ctx context.Context) ([]*storepb.RuleGroup, storage.Warnings, error) {
+	resp := &rulesServer{ctx: ctx}
+
+	if err := rr.proxy.Rules(&storepb.RulesRequest{
+		PartialResponseStrategy: storepb.PartialResponseStrategy_ABORT,
+	}, resp); err != nil {
+		return nil, nil, errors.Wrap(err, "proxy RuleGroups()")
```

Comment:
consider not captilazing error strings https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
hmm ... I wanted to stay consistent with https://github.com/thanos-io/thanos/blob/4240e3c743ba56969d32ad818a9208de0732f691/pkg/query/querier.go#L208

wdyt?